test/conformance/conformance_allocators.cpp

```diff
@@ -31,13 +31,15 @@ TEST_CASE("Allocator concept") {
     TestAllocator<oneapi::tbb::cache_aligned_allocator<void>>(Concept);
     TestAllocator<oneapi::tbb::tbb_allocator<void>>(Concept);
 
+  #ifndef __POWERPC__ // This test fails to build on Darwin PPC.
```

Comment:
This should test for `__powerpc__ && __APPLE__` since the test passes on Linux on PPC.

Reply:
@glaubitz I expected `__POWERPC__` to be Apple-specific, but sure, let us add `__APPLE__` then.

Reply:
You're right. gcc defines `__powerpc__` on Linux in lower-case only:

```bash
(sid_powerpc-dchroot)glaubitz@perotto:~$ echo | gcc -E -dM -|grep -i POWERPC
#define __powerpc__ 1
#define powerpc 1
#define __powerpc 1
(sid_powerpc-dchroot)glaubitz@perotto:~$
```

However, I would still check for `__APPLE__` as well to improve the readability of the code.

Reply:
Looking at the gcc sources, it seems that `__POWERPC__` is defined on at least BeOS as well:

```C
#if (defined (_ARCH_PPC)        /* AIX */                               \
     || defined (__powerpc__)   /* gcc */                               \
     || defined (__POWERPC__)   /* BEOS */                              \
     || defined (__ppc__)       /* Darwin */                            \
     || (defined (PPC) && ! defined (CPU_FAMILY)) /* gcc 2.7.x GNU&SysV */    \
     || (defined (PPC) && defined (CPU_FAMILY)    /* VxWorks */               \
         && CPU_FAMILY == PPC)
```

So, I would definitely check for both `__POWERPC__` and `__APPLE__` to be safe.

Reply:
@glaubitz Yes, you are right, it is better to consider that case. I have rebased now, adding __APPLE__ in the condition here and also moving to your code in GNU.cmake.